### PR TITLE
Buffer attribute types

### PR DESCRIPTION
**Pull request by rchadwic**
How do you feel about allowing bufferAttributes to use other data types? I've modified the renderer to check for different data types, and added the "normalized" setting to bufferAttributes. This allows us to use less memory for attributes that don't require full float32 precision. I've checked in an example that uses 8 bit uints for colors and 16 bit ints for normals.

### CODE REVIEW

**Comment by WestLangley** [comment]
Related: https://github.com/mrdoob/three.js/issues/8358

Do you mind removing the builds from the PR? See the Wiki article [How to Contribute to three.js](https://github.com/mrdoob/three.js/wiki/How-to-contribute-to-three.js).

Thanks for the example, too!

http://ci.threejs.org/api/pullrequests/8616/examples/webgl_buffergeometry_bufferDataTypes.html

**Comment by rchadwic** [comment]
Of course, sorry. I'll add a new commit to this PR. Question on this topic -  If I have several attributes with different DataViews on the same arrayBuffer, is that arrayBuffer stored in GPU memory multiple times?

File: src/core/BufferAttribute.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * @author mrdoob / http://mrdoob.com/
  */
 
-THREE.BufferAttribute = function ( array, itemSize ) {
+THREE.BufferAttribute = function ( array, itemSize, normalized ) {
```
Comment (mrdoob) [inline src/core/BufferAttribute.js]: I wouldn't add the extra parameter to the constructor by now.

File: src/renderers/WebGLRenderer.js [hunk 2]
```diff
@@ -970,6 +970,63 @@ THREE.WebGLRenderer = function ( parameters ) {
...
 				if ( geometryAttribute !== undefined ) {
 
+					var dataType = _gl.FLOAT;
+					var normalized = false;
+					if( geometryAttribute.array instanceof Float32Array ) {
+
+						dataType = _gl.FLOAT;
+
+					}
+					if( geometryAttribute.array instanceof Float64Array ) {
```
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js]: `} if else {`?

**Comment by mrdoob** [comment]
Seems like the PR has conflicts. Maybe it's because you deleted the builds?
I think you may need to rebase it.

**Comment by rchadwic** [comment]
Blah, sorry. Let's delete this PR and try again. I'll rebase off dev.

**Comment by mrdoob** [comment]
#8618

**Pull request closed without merging**